bundles/org.openhab.binding.boschindego/src/main/java/org/openhab/binding/boschindego/internal/IndegoController.java

```diff
@@ -0,0 +1,497 @@
+     * 
+     * @param command the control command to send to the device
+     * @throws IndegoAuthenticationException if request was rejected as unauthorized
+     * @throws IndegoInvalidCommandException if the command was not processed correctly
+     * @throws IndegoException if any communication or parsing error occurred
+     */
+    public void sendCommand(DeviceCommand command)
+            throws IndegoAuthenticationException, IndegoInvalidCommandException, IndegoException {
+        SetStateRequest request = new SetStateRequest();
+        request.state = command.getActionCode();
+        putRequestWithAuthentication(SERIAL_NUMBER_SUBPATH + this.getSerialNumber() + "/state", request);
+    }
+
+    /**
+     * Queries the predictive weather forecast.
+     * 
+     * @return the weather forecast DTO
+     * @throws IndegoAuthenticationException if request was rejected as unauthorized
+     * @throws IndegoException if any communication or parsing error occurred
+     */
+    public LocationWeatherResponse getWeather() throws IndegoAuthenticationException, IndegoException {
+        return getRequestWithAuthentication(SERIAL_NUMBER_SUBPATH + this.getSerialNumber() + "/predictive/weather",
+                LocationWeatherResponse.class);
+    }
+
+    /**
+     * Queries the predictive adjustment.
+     * 
+     * @return the predictive adjustment
+     * @throws IndegoAuthenticationException
+     * @throws IndegoException
+     */
+    public int getPredictiveAdjustment() throws IndegoAuthenticationException, IndegoException {
+        return getRequestWithAuthentication(
+                SERIAL_NUMBER_SUBPATH + this.getSerialNumber() + "/predictive/useradjustment",
+                PredictiveAdjustment.class).adjustment;
+    }
+
+    /**
+     * Sets the predictive adjustment.
+     * 
+     * @param adjust the predictive adjustment
+     * @throws IndegoAuthenticationException
+     * @throws IndegoException
+     */
+    public void setPredictiveAdjustment(final int adjust) throws IndegoAuthenticationException, IndegoException {
+        final PredictiveAdjustment adjustment = new PredictiveAdjustment();
+        adjustment.adjustment = adjust;
+        putRequestWithAuthentication(SERIAL_NUMBER_SUBPATH + this.getSerialNumber() + "/predictive/useradjustment",
+                adjustment);
+    }
+
+    /**
+     * Queries predictive moving.
+     * 
+     * @return predictive moving
+     * @throws IndegoAuthenticationException
+     * @throws IndegoException
+     */
+    public boolean getPredictiveMoving() throws IndegoAuthenticationException, IndegoException {
+        final PredictiveStatus status = getRequestWithAuthentication(
+                SERIAL_NUMBER_SUBPATH + this.getSerialNumber() + "/predictive", PredictiveStatus.class);
+        return status.enabled;
+    }
+
+    /**
+     * Sets predictive moving.
+     * 
+     * @param enable
+     * @throws IndegoAuthenticationException
+     * @throws IndegoException
```

Comment:
Please comment the 2 exceptions like you did for other methods.

Reply:
Fixed.